adit/core/utils/dicom_to_nifti_converter.py

```
@@ -45,9 +76,45 @@ def convert(self, dicom_folder: str | Path, output_folder: str | Path) -> None:
+                )
+            elif exit_code == DcmToNiftiExitCode.PARTIAL_CONVERSION:
+                logger.warning(f"Converted some but not all input DICOMs: {error_msg}")
+            elif exit_code == DcmToNiftiExitCode.RENAME_ERROR:
+                raise DcmToNiftiConversionError(f"Unable to rename files: {error_msg}")
+            elif exit_code == DcmToNiftiExitCode.UNSPECIFIED_ERROR or exit_code != 0:
+                raise DcmToNiftiConversionError(
+                    f"Unspecified error (exit code {exit_code}): {error_msg}"
+                )
+
+        except subprocess.SubprocessError as e:
+            raise ExternalToolError(f"Failed to execute dcm2niix: {e}")
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)